src/watershed.jl

```diff
@@ -0,0 +1,89 @@
+import Base.isless
+
+immutable PixelKey{CT}
+    val::CT
+    time_step::Int
+end
+isless{T}(a::PixelKey{T}, b::PixelKey{T}) = (a.val < b.val) || (a.val == b.val && a.time_step < b.time_step)
+
+"""
+```
+segments                = watershed(img, [markers])
+``` 
+Segments the image using watershed transform. Each basin formed by watershed transform corresponds to a segment.
+If you are using image local minimas as markers, consider using hmin_transform to avoid oversegmentation.
+    
+Parameters:  
+-    img            = input grayscale image
+-    markers        = an array marking the pixels where flooding should start
```

Comment:
Still not sufficiently explicit. How, exactly, should those pixels be marked? Is a `Vector{CartesianIndex{N}}` sufficient? How about a `Array{Bool,N}`? Finally, the docstring suggests this is an optional input but it doesn't appear to be optional (should we recommend a function for initializing `markers`?)